src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs

```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
 
         public class ParameterViewModel
         {
-            private readonly IParameterSymbol _parameter;
             private ChangeSignatureDialogViewModel _changeSignatureDialogViewModel;
 
-            public IParameterSymbol ParameterSymbol
-            {
-                get { return _parameter; }
-            }
+            public IParameterSymbol ParameterSymbol { get; }
 
             public ParameterViewModel(ChangeSignatureDialogViewModel changeSignatureDialogViewModel, IParameterSymbol parameter)
             {
                 _changeSignatureDialogViewModel = changeSignatureDialogViewModel;
-                _parameter = parameter;
+                ParameterSymbol = parameter;
             }
 
-            public string ParameterAutomationText
-            {
-                get { return $"{Type} {Parameter}"; }
-            }
+            public string ParameterAutomationText => $"{Type} {Parameter}";
 
             public string Modifier
             {
                 get
                 {
-                    // Todo: support VB
-                    switch (_parameter.RefKind)
+                    string CS()
                     {
-                        case RefKind.Out:
-                            return "out";
-                        case RefKind.Ref:
-                            return "ref";
-                        case RefKind.In:
-                            return "in";
+                        switch (ParameterSymbol.RefKind)
+                        {
+                            case RefKind.Out:
+                                return "out";
+                            case RefKind.Ref:
+                                return "ref";
+                            case RefKind.In:
+                                return "in";
+                        }
+
+                        if (ParameterSymbol.IsParams)
+                        {
+                            return "params";
+                        }
+
+                        if (_changeSignatureDialogViewModel._thisParameter != null &&
+                            ParameterSymbol == _changeSignatureDialogViewModel._thisParameter.ParameterSymbol)
+                        {
+                            return "this";
+                        }
+                        return string.Empty;
                     }
```

Comment:
not a fan of local functions here.  however, if you are going to use local functions, place at end of function after the final 'return' statement.